### ISSUE TITLE
running mocha shows yargs deprecation with latest yargs dependency

### ISSUE THREAD

**Issue by tripodsan** [mochajs/mocha#3742]
### Prerequisites
- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend avoiding the use of globally installed Mocha.

### Description
My project uses the latest `yargs@13.1.0`, which reports a deprecation warning when its parser is configured in the `package.json`. see https://github.com/yargs/yargs/pull/1262.

because mocha configures it's own yargs in its [package.json](https://github.com/mochajs/mocha/blob/master/package.json#L602-L606), running the tests will show the deprecation warning.

### Steps to Reproduce
example project: https://github.com/tripodsan/mochayargstest

`test.js`
```js
const yargs = require('yargs');
function runCli() {
  new yargs()
    .help().parse(['--help']);
}
describe('Test Yargs', () => {
  it('shows help', () => {
    runCli();
  })
});
```

1. run `npm test`
```
$ npm test

> mochayargs@1.0.0 test /Users/nobody/tmp/mochayargs
> mocha test.js

  Test Yargs
Configuring yargs through package.json is deprecated and will be removed in the next major release, please use the JS API instead.
Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]
```

2. run w/o mocha

```
$ node index.js
Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]
```
**Expected behavior:** 
no deprecation warning is shown from yargs.

**Actual behavior:**
deprecation warning is shown then running tests.

**Reproduces how often:** [What percentage of the time does it reproduce?]
always.

### Versions
```
    "mocha": "^6.0.0",
    "yargs": "^13.2.1"
```

### Additional Information
I think that mocha should use the new https://github.com/yargs/yargs/blob/master/docs/api.md#parserconfigurationobj instead of configuring the parser through the `package.json`

**Comment by plroebuck** [comment]
Asked for recommendation in yargs/yargs#1262 if upgrade to yargs-13 not possible.

**Comment by plroebuck** [comment]
yargs/yargs#1302

**Comment by boneskull** [comment]
I've discussed this w/ yargs team and it looks like they'll roll back dropping support of Node.js v6 in v13, so when that's released we can upgrade to v13 of yargs and use its recommended method of setting parser options.

**Comment by romainPrignon** [comment]
It looks like yargs re-introduce node v6 support in `v13.2.2`
https://github.com/yargs/yargs/compare/v13.2.1...v13.2.2

Shall we use https://github.com/yargs/yargs/blob/master/docs/api.md#parserconfigurationobj now ?

Is there a pull request for that already ?

**Comment by romainPrignon** [comment]
Thanks ! :+1: 
no more warning in test output now.

**Comment by KenEucker** [comment]
I am getting this warning now in my project using versions:

`"yargs": "^13.2.2"
 "mocha": "^6.0.2"`

It seems resolved for others, though?

**Comment by plroebuck** [comment]
@KenEucker fix was merged a week ago, whereas Mocha-6.0.2 more than 2 weeks ago.
PR will be part of next release.

**Comment by saramorsi** [comment]
I'm running into this same issue. How do I fix it? Using a later version of Mocha (e.g. 8) does not help.
